plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/HostedHandlerSupport.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javax.annotation.Nonnull;
+
+import org.sonatype.nexus.common.hash.HashAlgorithm;
+import org.sonatype.nexus.common.time.Clock;
+import org.sonatype.nexus.repository.content.InvalidContentException;
+import org.sonatype.nexus.repository.http.HttpResponses;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.Handler;
+import org.sonatype.nexus.repository.view.Response;
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import com.google.common.hash.HashCode;
+import org.joda.time.DateTime;
+
+import static org.sonatype.nexus.repository.http.HttpMethods.DELETE;
+import static org.sonatype.nexus.repository.http.HttpMethods.GET;
+import static org.sonatype.nexus.repository.http.HttpMethods.PUT;
+import static org.sonatype.nexus.repository.maven.internal.MavenPayload.ok;
+
+/**
+ * Maven hosted handler base.
+ *
+ * @since 3.0
+ */
+public abstract class HostedHandlerSupport<C extends Path>
```

Comment:
I recommend avoiding generics, except where isolated in helpers or where there is a clear pattern needed where generics simplifies things. More often than not it adds complexity instead of reducing it... so please keep that in mind.